### ISSUE TITLE
Defer loading option

### ISSUE THREAD

**Issue by booni3** [kdion4891/laravel-livewire-tables#14]
Hey,

Loving the package so far! Have you considered adding a deferred loading to the package? I cannot see any negatives to doing this even by default... but you could also provide a config for it.

I have implemented it by extending your table component as follows (the below method ensures pagination is not broken during loading)

```php
class TableComponent extends BaseTableComponent
{
    public $loaded = false;

    public function tableLoaded()
    {
        $this->loaded = true;
    }

    public function tableView()
    {
        return view('laravel-livewire-tables::table', [
            'columns' => $this->columns(),
            'models' => $this->loaded ?
                $this->models()->paginate($this->per_page)
                : $this->emptyModel(),
        ]);
    }

    private function emptyModel()
    {
        return $this->query()->paginate();
    }
}
```

Then at the top of the table livewire component
```html
<div wire:init="tableLoaded"></div>
```

What do you think?

**Comment by kejojedi** [comment]
what is the benefit of this

**Comment by booni3** [comment]
This is to ensure page load is not held up by data crunching. In the below example, I have set my speed to a fast 3g connection (It is running locally on valet so is actually slower than this still). Our table has about 2 million rows in total and we are showing 50 at a time. Laravel pagination is not particularly quick. Without defer loading we have a few seconds of nothing before anything loads.

![Kapture 2020-04-01 at 9 48 44](https://user-images.githubusercontent.com/20134485/78117916-2651bd80-73fe-11ea-8816-8a831155a29a.gif)

**Comment by kdion4891** [comment]
submit a PR pls.
